### PR TITLE
Be more receptive to offline environments

### LINKED ISSUE #19: It should be possible to disable remote repository fetching for offline use cases
In a disconnected environment it may still be convenient to use `eselect repository create` to automate the creation of a local ebuild repository.

Currently there does not appear to be a convenient way to skip the fetch of known repositories.

Suggestion: Both a command-line flag e.g. `--offline` and a configuration file option should exist to enable this use case.

### PR DESCRIPTION
**Pull request by Kangie**
Current behaviour is to update the cached repositories.xml whenever an operation is performed, regardless of whether we _need_ that information to continue (e.g. creating a new local repo, adding a repo by uri, removing an existing repo).

This commit changes this behaviour to only update the cache when performing operations that use the cache and adds a new `OFFLINE_MODE` configuration variable that skips fetching repositories.xml entirely.

Closes #19

### CODE REVIEW

**Comment by Kangie** [comment]
the TL;DR here is that now only `enable` and `list` try to update the repositories.xml by default (this is in line with existing documentation), and that can be overridden by OFFLINE_MODE.

Rationale:

- Disabling means you know the cached number or are using name
- Removing is the same
- Creation is an offline action
- Add provides `name`, `sync-type`, and `uri`

This came out of my recent experiences trying to use something as basic as `eselect repository create` in a firewalled environment; it's a really convenient helper that shouldn't rely on a webrequest succeeding to perform most functions.

**Comment by Kangie** [comment]
It's not quite there yet, as there's an edge case around 'what do we do if there isn't a cached xml', but I figured it was worth raising for comment while I iron the rough edges out.

Edit: What we do is create an empty XML file if we don't have a cached version to fall back on in offline mode, with predictable results - list only shows local repositories (though doesn't get remote URIs from repos.conf, hm, potential enhancement), and enable does nothing useful because we only know about local repositories.

File: repository.eselect.in [hunk 1]
```diff
@@ -45,20 +45,31 @@ get_config() {
 update_cache() {
 	local refts
 
-	if [[ -f ${REMOTE_LIST_CACHE} ]]; then
-		refts=$(( $(date '+%s') - REMOTE_LIST_REFRESH ))
-		[[ $(find "${REMOTE_LIST_CACHE}" -newermt "@${refts}") ]] && return
-	fi
+	if ! [ "${OFFLINE_MODE}" = "true" ]; then
```
Comment (mgorny) [inline repository.eselect.in:48]: eselect is bash.

File: repository.eselect.in [hunk 2]
```diff
@@ -45,20 +45,31 @@ get_config() {
...
+		touch "${REMOTE_LIST_CACHE}"
+	else
+		# We need some sort of valid XML for the helper to work
+		if [[ -f ${REMOTE_LIST_CACHE} ]]; then
+			cat > ${REMOTE_LIST_CACHE} <<-EOF || die
+				<?xml version="1.0" ?>
+				<metadata>
+				</metadata>
+				EOF
+		fi
```
Comment (mgorny) [inline repository.eselect.in:71]: No, just don't. Writing junk to workaround a bug is not a solution.

**Comment by thesamesam** [comment]
(wrong button, meant to run CI)

**Comment by maffblaster** [comment]
What are we waiting on here? Code revisions from @Kangie?

**Comment by mgorny** [comment]
Yes. The comments weren't addressed.
